rsocket-core/src/main/java/io/rsocket/internal/BaseDuplexConnection.java

```diff
@@ -11,7 +11,7 @@
   protected UnboundedProcessor sender = new UnboundedProcessor();
 
   public BaseDuplexConnection() {
-    onClose.doFinally(s -> doOnClose()).subscribe();
+    onClose.subscribe(null, t -> doOnClose(), this::doOnClose);
```

Comment:
Does this still run for cancellation?

Reply:
I dont think it should at all. In this case, we wait for onError or onComplete